be/src/connector/hive_connector.cpp

```diff
@@ -579,6 +579,95 @@ HdfsScanner* HiveDataSource::_create_paimon_jni_scanner(const FSOptions& options
     return scanner;
 }
 
+HdfsScanner* HiveDataSource::_create_hive_jni_scanner(const FSOptions& options) {
+    const auto& scan_range = _scan_range;
+    const auto* hdfs_table = dynamic_cast<const HdfsTableDescriptor*>(_hive_table);
+    auto* partition_desc = hdfs_table->get_partition(scan_range.partition_id);
+    std::string partition_full_path = partition_desc->location();
+
+    std::string required_fields;
+    for (auto const& slot : _materialize_slots) {
+        required_fields.append(slot->col_name());
+        required_fields.append(",");
+    }
+    required_fields = required_fields.substr(0, required_fields.size() - 1);
+
+    std::string nested_fields;
+    for (auto slot : _materialize_slots) {
+        const TypeDescriptor& type = slot->type();
+        if (type.is_complex_type()) {
+            build_nested_fields(type, slot->col_name(), &nested_fields);
+        }
+    }
+    if (!nested_fields.empty()) {
+        nested_fields = nested_fields.substr(0, nested_fields.size() - 1);
+    }
+
+    std::string data_file_path;
+    if (scan_range.relative_path.empty()) {
+        data_file_path = "";
```

Comment:
this could be wrong. 

think about when data file path is empty, it's almost error.